### PR TITLE
Syslog: add necessary options to cover the major usability

### LINKED ISSUE #32314: Syslog: add necessary options to cover the major usability
### Description

As GELF will be removed from Keycloak, we should have an adequate replacement for it. The most suitable candidate is Syslog, as it provides the most crucial ability - centralized log management.

In order to be, at least, as usable as GELF, we should also add other options that users can configure.

After some investigation, these options should be included in the configuration:

- [**Log level** ](https://quarkus.io/guides/logging#quarkus-log-logging-log-config_quarkus-log-syslog-level)- to be able to specify log level only for Syslog
- [**Syslog type**](https://quarkus.io/guides/logging#quarkus-log-logging-log-config_quarkus-log-syslog-syslog-type) - whether Syslog should use RFC5424, or RFC3164
- [**Max message length**](https://quarkus.io/guides/logging#quarkus-log-logging-log-config_quarkus-log-syslog-max-length) - new Quarkus option for specifying the maximum length of the message that can be sent

### PR DESCRIPTION
**Pull request by mabartos**
- Closes #32314 

### Docs preview
image

### CODE REVIEW

**Comment by mabartos** [comment]
@keycloak/cloud-native-maintainers Could you please take a look? Thank you.

File: quarkus/config-api/src/main/java/org/keycloak/config/LoggingOptions.java [hunk 1]
```diff
@@ -158,40 +161,58 @@ public String toString() {
...
+            .category(OptionCategory.LOGGING)
+            .expectedValues(Arrays.stream(SyslogHandler.SyslogType.values()).map(f -> f.toString().toLowerCase()).toList())
+            .description("Set the Syslog type used to format the sent message.")
+            .defaultValue(SyslogHandler.SyslogType.RFC5424.toString().toLowerCase())
+            .build();
+
+    public static final Option<Level> LOG_SYSLOG_LEVEL = new OptionBuilder<>("log-syslog-level", Level.class)
+            .category(OptionCategory.LOGGING)
+            .defaultValue(Level.ALL)
+            .description("Set the log level specifying which message levels will be logged by the Syslog logger. Message levels lower than this value will be discarded.")
```
Comment (vmuzikar) [inline quarkus/config-api/src/main/java/org/keycloak/config/LoggingOptions.java:178]: Should we perhaps mention, if/how this plays with `log-level`? Which of these take precedence?

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/LoggingPropertyMappers.java [hunk 2]
```diff
@@ -234,6 +251,25 @@ private static Level toLevel(String categoryLevel) throws IllegalArgumentExcepti
...
+                .orElseThrow(() -> new IllegalArgumentException("You need to provide a valid Syslog type"));
+
+        return Optional.of(
+                switch (syslogType) {
+                    case RFC5424:
+                        yield "2048";
+                    case RFC3164:
+                        yield "1024";
+                });
+    }
```
Comment (vmuzikar) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/LoggingPropertyMappers.java:271]: Just out of curiosity. Do we really need this logic? Isn't it handled by Quarkus?

File: quarkus/config-api/src/main/java/org/keycloak/config/LoggingOptions.java [hunk 3]
```diff
@@ -158,40 +161,58 @@ public String toString() {
...
             .build();
 
+    public static final Option<String> LOG_SYSLOG_TYPE = new OptionBuilder<>("log-syslog-type", String.class)
+            .category(OptionCategory.LOGGING)
+            .expectedValues(Arrays.stream(SyslogHandler.SyslogType.values()).map(f -> f.toString().toLowerCase()).toList())
+            .description("Set the Syslog type used to format the sent message.")
+            .defaultValue(SyslogHandler.SyslogType.RFC5424.toString().toLowerCase())
+            .build();
+
+    public static final Option<Level> LOG_SYSLOG_LEVEL = new OptionBuilder<>("log-syslog-level", Level.class)
```
Comment (vmuzikar) [inline quarkus/config-api/src/main/java/org/keycloak/config/LoggingOptions.java:175]: The level seems like a generic option applicable to all log handlers. Would it make sense as a follow-up to add a similar level option to other handlers as well (assuming they support it)?

**Comment by mabartos** [comment]
I tried to experiment a little bit with the log levels for syslog and root, and it seems there's a bigger issue on the Quarkus side. I'll verify it.

**Comment by mabartos** [comment]
## Current state

- **Added the `log-syslog-type` option** - possibility to change the message format - RFC3164 still used in many apps
- **Added the `log-syslog-message-length` option** - to mimic GELF and tailor more to the users' needs
- **NO log-level for Syslog now** - Needs to be done in a follow-up issue with other log handlers levels - deeper issue
- **NO custom logic for the default message length** - need to rely on Quarkus testing as it's not even propagated to the quarkus property; so no comprehensive testing - the defaults are set in SyslogHandler of JBoss LogManager (outside of Quarkus)

**Comment by mabartos** [comment]
@vmuzikar @shawkins Could you please review it? Thanks

**Review by shawkins — APPROVED** [review]
LGTM, thanks @mabartos

**Review by vmuzikar — APPROVED** [review]
LGTM, nice job @mabartos.

**Pull request merged**